src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -12,11 +18,38 @@ public MachineDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert machine " + machine, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Machine> criteriaQuery = criteriaBuilder.createQuery(Machine.class);
+            Root<Machine> root = criteriaQuery.from(Machine.class);
+            Predicate predicate = criteriaBuilder.gt(root.get("age"), age);
```

Comment:
u need find year in DB and this year must be less than year you got , but u have only age, so find year